test/e2e/install_test.go

```diff
@@ -154,8 +154,12 @@ var _ = Describe("Operator Install", func() {
 				PackageName: pkgName,
 			}
 
-			// Delete the catalog first
+			By("deleting the catalog first")
 			Expect(c.Delete(ctx, operatorCatalog)).To(Succeed())
+			Eventually(func(g Gomega) {
+				err := c.Get(ctx, types.NamespacedName{Name: operatorCatalog.Name}, &catalogd.Catalog{})
+				g.Expect(errors.IsNotFound(err)).To(BeTrue())
+			}).Should(Succeed())
```

Comment:
We could retry on failed deletes until we get a 404, this allows us to recover from the initial delete call failure.
```suggestion
			Eventually(func(g Gomega) {
				err := c.Delete(ctx, operatorCatalog)
				g.Expect(errors.IsNotFound(err)).To(BeTrue())
			}).Should(Succeed())
```

Reply:
Bit of a nit btw.

Reply:
Could do
```
g.Expect(errors.IsNotFound(c.Delete(ctx, operatorCatalog))).To(BeTrue())
```

Reply:
I like that! We have one place related to catalog deletion. I'll update it too to be consistent.

Reply:
I don't think I'd do it this way. We expect a delete always to succeed. If you had a concern about it succeeding, you _could_ wrap the delete itself in an eventually (although I wouldn't).

This pattern of
1. Do an action that is always expected to succeed
2. Wait until result of action is eventually visible

is fairly standard for k8s e2e tests.

Reply:
That also makes sense.

@ncdc @awgreene given that it works as we want both ways and we already have a pattern of delete & wait elsewhere - I'll keep it as currently is.